README.rst

```diff
@@ -97,6 +97,17 @@ To use this with your project you need to follow these steps:
         }
 
         LOG_OUTGOING_REQUESTS_DB_SAVE = True # save logs enabled/disabled based on the boolean value
+        LOG_OUTGOING_REQUESTS_DB_SAVE_BODY = True # save request/response body
+        LOG_OUTGOING_REQUESTS_EMIT_BODY = True # log request/response body
+        LOG_OUTGOING_REQUESTS_CONTENT_TYPES = [
+                "text/*",
+                "application/json",
+                "application/xml",
+                "application/soap+xml",
+        ] # save request/response bodies with matching content type
```

Comment:
as discussed in the office - this would probably benefit from a dataclass so you can do things like:

```py
LOG_OUTGOING_REQUESTS_CONTENT_TYPES = [
    ContentType(pattern="text/xml", default_encoding="iso-8859-3"),
    ContentType(pattern="text/*", default_encoding="utf-8"),
    ContentType(pattern="application/json", default_encoding="utf-8"),
    ...
]
```

then algorithm-wise you:

1. extract the content type of the body (request or response)
2. for every item in `settings.LOG_OUTGOING_REQUESTS_CONTENT_TYPES`
    1. check if there's an exact match
    2. if not, check if there's a pattern match (using the wildcard)
3. if there's no match -> don't log it
4. if there's a match
    1. try to get the charset from the content-type header
    2. if there's no charset, fall back to the `ContentType.default_encoding`
5. Log the body

Somewhere in between there's also the check if settings/configuration and the body size

We also discussed that you might provide a setting/escape hatch to derive the encoding, something along the lines of:

```py
def extract_encoding_for_body(content_type: str, body: bytes) -> str:
    ...  # default implementation is part of the above algorithm


LOG_OUTGOING_REQUESTS_EXTRACT_ENCODING = "log_outgoing_requests.utils.extract_encoding_for_body"
```

so the default setting value should be the library function, but projects can provide their own function. In the case of XML for example, you'd want to parse the first line and get the value from the `<?xml encoding="...">` part, but that's for projects to figure out.